lucetc/src/decls.rs

```diff
@@ -117,18 +117,31 @@ impl<'a> ModuleDecls<'a> {
         for ix in 0..decls.info.functions.len() {
             let func_index = UniqueFuncIndex::new(ix);
 
+            // Get the name for this function from the module names section, if it exists.
+            // Because names have to be unique, we append the index value (ix) to the name.
+            fn custom_name_for<'a>(
+                ix: usize,
+                func_index: UniqueFuncIndex,
+                decls: &mut ModuleDecls<'a>,
+            ) -> Option<String> {
+                decls
+                    .info
+                    .function_names
+                    .get(&func_index)
+                    .cloned()
+                    .map(|s| format!("{}_{}", s, ix))
+            }
+
             fn export_name_for<'a>(
                 func_ix: UniqueFuncIndex,
                 decls: &mut ModuleDecls<'a>,
             ) -> Option<String> {
                 let export = decls.info.functions.get(func_ix).unwrap();
-
```

Comment:
This is an etiquette question tangential to the PR at hand, but because the _Opinions About Whitespace Haver_ has logged on, I figured it was worth asking so as not to step on toes: Should I stop myself from making _technically unrelated_ changes like this in future PR's?

Reply:
I personally loosely prefer whitespace changes (in the form of trailing spaces or spaceful newlines) to be a distinct commit if not a different PR, but I'm curious about other opinions! (I tend to preserve whitespace except in situations of trailing whitespace, so I've never asked either)